pyfakefs/fake_filesystem.py

```diff
@@ -1480,8 +1530,8 @@ def components_to_path():
         for component in path_components:
             if not isinstance(current_dir, FakeDirectory):
                 return components_to_path()
-            dir_name, current_dir = self._directory_content(
-                current_dir, component)
+            dir_name, current_dir = self._directory_content(  # type: ignore
```

Comment:
Incompatible types in assignment (expression has type "Union[FakeFile, FakeDirectory, None]", variable has type "FakeDirectory") (same below), has to be fixed